src/orquestra/sdk/schema/ir.py

```diff
@@ -201,6 +217,10 @@ class TaskDef(BaseModel):
     # An empty list [] means a Task with no parameters
     parameters: t.Optional[t.List[TaskParameter]]
 
+    # Statically inferred from the task function. See also `TaskOutputMetadata`'s
+    # docstring.
+    output_metadata: TaskOutputMetadata
```

Comment:
I saw you commented on this after I started reviewing.

I think we'll need to be very explicit about this. We don't want to delay folks upgrading, and a breaking change may put folks off.
On the other hand, it might be better to rip the band-aid off now.

Reply:
Yeah, ripping band-aids sucks. Let's consider what other options we have:

1. Don't add `TaskDef.output_metadata` to the IR at all, and figure out some other way get this information in the CLI code (or rework CLI so it doesn't need this at all). 
    * Con 1: this still doesn't guard our users from bugs related to interacting with old workflow defs because I'm changing how `TaskInvocation.output_ids` are populated anyway (both on the producing and consuming side). 
    * Con 2: the CLI code might end up hacky, relying on even more implicit assumptions.
2. Add `TaskDef.output_metadata` to the IR, but make it optional. Old workflow defs get `None`, and the code that requires this information raises an exception telling the user this workflow is too old. `orq task results` is the only functionality I can think of; perhaps more can come in the future